From 963c461a188bb03ce7a3264768acd60c1b6c1973 Mon Sep 17 00:00:00 2001
From: Srikanth Srungarapu <ssrungarapu@cloudera.com>
Date: Wed, 11 Mar 2015 11:31:56 -0700
Subject: [PATCH 097/197] CLOUDERA-BUILD Revert "HBASE-10471 Remove HTD.isAsyncLogFlush() from trunk"

Reason: Product Requirement (Compatibility)
Author: Srikanth Srungarapu
Ref: CDH-24305

This reverts commit 3cc587b961f4630d85b0b70c745f4a74675fc92c.

Conflicts:
	hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/wal/TestDurability.java

Change-Id: I02a7642f35a90d79f19e42f9eb4bcd62e2a12e42
---
 .../org/apache/hadoop/hbase/HTableDescriptor.java  |   28 ++++++++++++++++++++
 1 files changed, 28 insertions(+), 0 deletions(-)

diff --git a/hbase-client/src/main/java/org/apache/hadoop/hbase/HTableDescriptor.java b/hbase-client/src/main/java/org/apache/hadoop/hbase/HTableDescriptor.java
index 36a1252..b874156 100644
--- a/hbase-client/src/main/java/org/apache/hadoop/hbase/HTableDescriptor.java
+++ b/hbase-client/src/main/java/org/apache/hadoop/hbase/HTableDescriptor.java
@@ -672,6 +672,34 @@ public class HTableDescriptor implements WritableComparable<HTableDescriptor> {
   }
 
   /**
+   * Check if async log edits are enabled on the table.
+   *
+   * @return true if that async log flush is enabled on the table
+   * @deprecated Since 0.96 we no longer have an explicity deferred log flush/sync functionality.
+   * Use {@link #getDurability()}.
+   */
+  public synchronized boolean isAsyncLogFlush() {
+    return getDurability() == Durability.ASYNC_WAL;
+  }
+
+  /**
+   * This is used to allowing the log edits syncing to the file system. Everytime
+   * an edit is sent to the server it is first sync'd to the file system by the
+   * log writer. This sync is an expensive operation and thus can be deferred so
+   * that the edits are kept in memory until the background async writer-sync-notifier
+   * threads do the sync and not explicitly flushed for every edit.
+   * <p>
+   * NOTE:- This option might result in data loss if the region server crashes
+   * before these pending edits in memory are flushed onto the filesystem.
+   * </p>
+   *
+   * @param isAsyncLogFlush
+   */
+  public synchronized void setAsyncLogFlush(final boolean isAsyncLogFlush) {
+    this.setDurability(isAsyncLogFlush ? Durability.ASYNC_WAL : DEFAULT_DURABLITY);
+  }
+
+  /**
    * Sets the {@link Durability} setting for the table. This defaults to Durability.USE_DEFAULT.
    * @param durability enum value
    */
-- 
1.7.0.4

